lib/swagger/blocks/nodes/root_node.rb

```diff
@@ -50,10 +50,18 @@ def tag(inline_keys = nil, &block)
 
         def server(inline_keys = nil, &block)
           raise NotSupportedError unless is_openapi_3_0?
+
           self.data[:servers] ||= []
           self.data[:servers] << Swagger::Blocks::Nodes::ServerNode.call(version: version, inline_keys: inline_keys, &block)
         end
 
+        def extension(name, inline_keys = nil, &block)
+          raise NotSupportedError unless is_openapi_3_0?
+
+          self.data["x-#{name}".to_sym] ||= []
```

Comment:
Sorry for the late reply @swiknaba. What do you think about the name being the whole key? The name would be 
"x-foo-bar" rather than "foo-bar". I think that limiting the "magic" here is probably a good idea. If someone is grepping for "x-foo-bar" it would be good if they could easily find it.

Reply:
^^ no problem, I'm super busy at the moment as well.

My original thoughts were two:
* call it with `x :foo` to generate `"x-foo"`, but it is now anyway called `extension`
* prevent the user from generating invalid OAS3 JSON -> do you think, we should raise an error, if the name does not start with an `x` if we use a bit less magic as you suggest?

Thinking about it, I would probably not raise an error and leave it to the user to validate the output. I agree, grepping for "x-foo-bar" is a very common thing to do, and one should find it that way. I'll bump the version to not break my own production code.